tidy3d/components/eme/simulation.py

```diff
@@ -231,6 +232,12 @@ class EMESimulation(AbstractYeeGridSimulation):
     _freqs_not_empty = validate_freqs_not_empty()
     _freqs_lower_bound = validate_freqs_min()
 
+    @pd.validator("size", always=True)
+    def _validate_size(cls, val):
+        """An EME simulation must be fully 3D."""
+        if any(val == 0):
+            raise SetupError("'EMESimulation' cannot have any component of 'size' equal to zero.")
```

Comment:
hm, shouldn't this return `val` if it works? I'm a bit confused

Reply:
good catch. The validator was overwritten by another validator with the same name below, so it was not being run. I added a few more tests to catch this

Reply:
often, other errors were being raised, say when the mode solver monitors were not 2D. but now I test zero size in transverse direction too

Reply:
ah ok 👍  approving